plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+    uid: "a1b2c3d4-1234-5678-90ab-f0e1d2c3b4a5"
+    controller: true
+    blockOwnerDeletion: true
+spec:
+  # The parent MultigresCluster controller passes down the
+  # images relevant to this CR's children (Shard).
+  images:
+    multipooler: "multigres/multigres:latest"
+    postgres: "postgres:15.3"
+
+  partitioning:
+    shards: 2
+
+# The multigresCluster controller watches and resolves the `deploymentTemplate` entries into the 'shardTemplate'.
+  shardTemplate:
+    pools:
+      - type: "replica"
+        cell: "us-west-2"
+        replicas: 2
+        dataVolumeClaimTemplate:
+          accessModes: ["ReadWriteOnce"]
+          resources:
+            requests:
+              storage: "500Gi"
+        postgres:
+          resources:
+            requests:
+              cpu: "4"
+              memory: "8Gi"
+        multipooler:
+          resources:
+            requests:
+              cpu: "1"
+              memory: "512Mi"
+
+      - type: "readOnly"
+        cell: "us-east-1"
+        replicas: 1
+        dataVolumeClaimTemplate:
+          accessModes: ["ReadWriteOnce"]
+          resources:
+            requests:
+              storage: "500Gi"
+        postgres:
+          resources:
+            requests:
+              cpu: "2"
+              memory: "4Gi"
+        multipooler:
+          resources:
+            requests:
+              cpu: "500m"
+              memory: "256Mi"
+            limits:
+              cpu: "1"
+              memory: "512Mi"
+status:
+  conditions:
+  - type: Available
+    status: "True"
+    lastTransitionTime: "2025-10-21T10:37:00Z"
+    message: "All shards are healthy"
+  shards: 2
+  readyShards: 2
+  ```
+
+  ## Shard CR - Read-Only Child of TableGroup
+
+```yaml
+# This child CR is created by the 'TableGroup' controller
+# for 'production-db-orders-tg'. This is shard "0" of 2.
+#
+# The TableGroup controller copies its own 'spec.images'
+# and its 'spec.shardTemplate.pools' into this CR's spec.
+#
+apiVersion: multigres.com/v1alpha1
+kind: Shard
+metadata:
+  name: "production-db-orders-tg-0"
+  namespace: multigres
+  creationTimestamp: "2025-10-21T10:35:00Z"
+  generation: 1
+  resourceVersion: "12399"
+  uid: "e5f6a7b8-1234-5678-90ab-f0e1d2c3b4a8"
+  labels:
+    multigres.com/cluster: "example-multigres-cluster"
+    multigres.com/database: "production_db"
+    multigres.com/table-group: "orders_tg"
+    multigres.com/shard: "0"
+  ownerReferences:
+  - apiVersion: multigres.com/v1alpha1
+    kind: TableGroup
+    name: "production-db-orders-tg"
+    uid: "d4e5f6a7-1234-5678-90ab-f0e1d2c3b4a7"
+    controller: true
+    blockOwnerDeletion: true
+spec:
+  # This images struct is copied from the parent 'TableGroup'.
+  # The Shard controller will read these values to build pods.
+  images:
+    multipooler: "multigres/multigres:latest"
+    postgres: "postgres:15.3"
+
+    
+  # The 'pools' block is a direct copy of the
+  # 'shardTemplate.pools' from the parent TableGroup.
```

Comment:
This comment is a bit inaccurate. This is true not just for pools, but also the images stanza which is a direct copy from the TableGroup.

I think there was a clear reason why we wanted to have this nested structure, but it's not very clear why we need this CRD. Could you perhaps provide some example config that would make use of this nested structure of TableGroup -> Shard?